ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -163,6 +170,36 @@ data ChainSyncLoPBucketConfig
     -- | Enable the leaky bucket.
     ChainSyncLoPBucketEnabled ChainSyncLoPBucketEnabledConfig
 
+-- | Configuration of ChainSync Jumping
+data CSJConfig
+  =
+    -- Disable ChainSync Jumping. All clients will fully synchronize with
+    -- the chain of its peer.
+    CSJDisabled
+  |
+    -- | Enable ChainSync Jumping
+    CSJEnabled CSJEnabledConfig
+
+data CSJEnabledConfig = CSJEnabledConfig {
+  -- | The _ideal_ size for ChainSync jumps. Note that the algorithm
+  -- is best-effort: there might not be exactly `csjcJumpSize` slots between two
+  -- jumps, depending on the chain.
+  --
+  -- There can be a few less slots between jumps if there is not a block exactly
+  -- at the boundary. Jumps are often made when a block is announced after the
+  -- jump boundary.
+  --
+  -- There can be even less slots if a dynamo is elected and it requires an
+  -- initial jump regardless of how far we are from the next jump boundary.
+  --
+  -- csjcJumpSize must be greater than 0 and smaller or equal to the genesis
+  -- window size. The larger the jump size, the less jumps are made and peers
+  -- are less involved in the syncing. A jump size as large as the genesis
+  -- window has a higher change that dishonest peers can delay syncing by a
+  -- small margin (around 2 minutes per dishonest peer with mainnet parameters).
+  csjcJumpSize       :: SlotNo
```

Comment:
Either annotate this with `!` or convert this to a newtype

Reply:
[Made it a newtype](https://github.com/IntersectMBO/ouroboros-consensus/pull/1033/files#r1605023245).